lib/metadata_json_lint.rb

```diff
@@ -60,6 +66,19 @@ def parse(metadata)
         error_state = true
       end
       dep_names << dep['name']
+
+      # Open ended dependency
+      # From: https://docs.puppet.com/puppet/latest/reference/modules_metadata.html#best-practice-set-an-upper-bound-for-dependencies
+      begin
+        if dep['version_requirement'].nil? || SemanticPuppet::VersionRange.parse(dep['version_requirement']).end == SemanticPuppet::Version::MAX
+          puts "Warning: Dependency #{dep['name']} has an open ended dependency version requirement #{dep['version_requirement']}"
```

Comment:
Lines 73/74 are too long for rubocop, they should be shortened or split over multiple lines.

There are some further errors due to the introduced complexity, you might want to split this whole begin/rescue block into methods.